crates/uv-interpreter/src/find_python.rs

```diff
@@ -125,6 +125,20 @@ fn find_python(
     selector: PythonVersionSelector,
     cache: &Cache,
 ) -> Result<Option<Interpreter>, Error> {
+    // Validate the requested version.
+    match selector {
+        PythonVersionSelector::Major(major) if major < 3 => {
+            return Err(Error::UnsupportedPython(major.to_string()));
+        }
+        PythonVersionSelector::MajorMinor(major, minor) if (major, minor) < (3, 8) => {
+            return Err(Error::UnsupportedPython(format!("{major}.{minor}")));
+        }
+        PythonVersionSelector::MajorMinorPatch(major, minor, patch) if (major, minor) < (3, 8) => {
+            return Err(Error::UnsupportedPython(format!("{major}.{minor}.{patch}")));
+        }
```

Comment:
I think this is _only_ true for Python 3.7? Python 3.6 and lower don't work at all, at least on Linux -- we raise an error in the script.

Reply:
I can move this elsewhere.... I have to do something like this if we want to retain the behavior we had before, because I removed the `<= Some(2)` check, which will now never fire anyway (because we'll hit the `-I` error path, not the `UnsupportedVersion` error path).